File: final-client/src/app/models/post.ts
import { Time } from "@angular/common";

export class Post {
    postId?: string;
    body?: string;
    postDate?: Date;
    username?: string;

    constructor(id?: string, body?: string, postDate?: Date, username?: string) {
        this.postId = id;
        this.body = body;
        this.postDate = postDate;
        this.username = username;
    }
}

File: final-api/Models/Post.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using System;
using final_api.Models;


namespace final_api.Models;

public class Post
{
    public int PostId { get; set; }

    [Required]
    public string Body { get; set; }
    [Required]
    public DateTime PostDate { get; set; }
    [Required]
    public string Username { get; set; }
}